### ISSUE TITLE
Provide lists of lint / pycodestyle rules that Black makes obsolete

### ISSUE THREAD

**Issue by nixjdm** [psf/black#429]
Information Request / Addition to README

This is related to https://github.com/ambv/black/issues/259#issuecomment-392242976, but I have the opposite aim as that thread. For me, my editor shows everything flake8 is catching for me, but many things I'm intentionally not worrying about because I know Black will handle it. I want to continue to have flake8 tell me valuable things, but never bother me for anything Black handles. They are distractions. I can add all of these rules to my `.flake8`, which works great, but I've been only finding them as I go.

Is there a complete list of Black-made obsolete pycodestyle rules? I know only a subset of [these rules](http://pycodestyle.pycqa.org/en/latest/intro.html#error-codes) are made obsolete.

I'd love to just drop that list in my `.flake8` and never be bothered by the obsolete rules again. I suggest adding this list to the README.

**Comment by jgirardet** [comment]
The black plugin fur sublimetext has this addon, if it may help

**Comment by zsol** [comment]
Sounds like a reasonable thing we could document in the README :) Care to put together a PR?

**Comment by nixjdm** [comment]
@zsol Not at all, if you don't mind waiting for me to find the time! I'm a little busy at the moment, but I may free up in a week or two.

Additionally, adding all of these to an explicit tests would help ensure the rules stay obsolete, and help justify individual rules being on the list. Though the tests may also be redundant if they're already covered. I suspect most of them are. Would you like that too?

**Comment by nixjdm** [comment]
@jgirardet Thanks for that list! Do we have any idea if it's 'complete'?

**Comment by jgirardet** [comment]
We should ask to @kaste

**Comment by kaste** [comment]
It's not complete bc I ignored all tab warnings.

**Comment by tony** [comment]
We may also have to add W503 to this list: https://github.com/ambv/black/issues/52

**Comment by chdsbd** [comment]
Thanks for creating this issue. This is the list I've come up with:

```
E101
E111
E114
E115
E116
E117
E121
E122
E123
E124
E125
E126
E127
E128
E129
E131
E133
E2
E3
E5
E701
E702
E703
E704
W1
W2
W3
W503
W504
```

I wouldn't mind creating tests for this, but I'm not sure if they belong in or would be accepted in the Black repo. Maybe we should create a separate project?

**Comment by nixjdm** [comment]
@chdsbd, thank you for your additional list. It differs from https://github.com/kaste/SublimeLinter-addon-black-for-flake/blob/master/plugin.py. I haven't checked every rule in both lists, the I'd argue the true list is probably a combination of the two, maybe plus more (e.g. E223/E224, I assume). It's a sufficiently long list, and hard enough to keep in my head, that tests make a lot of sense to me.

IMO this list should be rules Black makes obsolete because it makes a decisive choice, not necessarily because it fixes it. If Black actively chooses to break a rule, it is still something I want to ignore. Let me pick on a rule in your list, but not Sublime's, to illustrate.

- **W503 line break before binary operator**: Black tries to prevent this, but will sometimes actively break this rule if there's enough complexity. Either way, it's controlled by Black, not the user, so it should be ignored. Sublime's omission is right that Black doesn't always *fix* it, but it does control it, which I think is the better criterion.

**Comment by ichard26** [comment]
Ping! I think PR #1371 pretty much resolves this issue. Although, I should look into why the list above has *a lot* of ignore rules.

**Comment by kaste** [comment]
As for SublimeLinter, this list here is applied as a post filter.  T.i. you can have a normal flake8 configuration for example for CI, only in the editor then a lot of errors are suppressed iff you're running the black formatter e.g. on save. That's why you have a minimal list for where black and flake8 just disagree, and the maximal list provided here where black is a fixer for flake8 errors.

**Comment by hauntsaninja** [comment]
I think we can close this. Feel free to open PRs against https://black.readthedocs.io/en/stable/guides/using_black_with_other_tools.html if that page can be made better
